### ISSUE TITLE
Running on not-Linux results in constant logcontext spam

### ISSUE THREAD

**Issue by jryans** [matrix-org/synapse#6055]
### Description

I am running Synapse on macOS and getting constant (many times per second) log messages like:

```
2019-09-18 14:57:24,338 - synapse.logging.context - 362 - WARNING - GET-16 - Called stop on logcontext GET-16 without calling start
2019-09-18 14:57:24,339 - synapse.logging.context - 362 - WARNING - GET-16 - Called stop on logcontext GET-16 without calling start
2019-09-18 14:57:24,340 - synapse.logging.context - 362 - WARNING - GET-16 - Called stop on logcontext GET-16 without calling start
2019-09-18 14:57:25,219 - synapse.logging.context - 362 - WARNING - update_client_ips-13 - Called stop on logcontext update_client_ips-13 without calling start
2019-09-18 14:57:25,221 - synapse.logging.context - 362 - WARNING - update_client_ips-13 - Called stop on logcontext update_client_ips-13 without calling start
2019-09-18 14:57:25,223 - synapse.logging.context - 362 - WARNING - update_client_ips-13 - Called stop on logcontext update_client_ips-13 without calling start
2019-09-18 14:57:25,285 - synapse.logging.context - 362 - WARNING - handle_presence_timeouts-7 - Called stop on logcontext handle_presence_timeouts-7 without calling start
```

For a longer log snippet, see [this gist](https://gist.github.com/jryans/6e4d8e281d637c5006599c6e774b65f7). 

@richvdh took an initial look and suggested there's something that would likely fail on macOS that causes this.

### Version information

- **Homeserver**: local on my development machine

If not matrix.org:

- **Version**: Synapse/1.3.1 (b=anoa/msc2290,cd6f3a89d,dirty)

- **Install method**: `pip install` from source

- **Platform**: macOS 10.14.6

**Comment by richvdh** [comment]
we have [this code](https://github.com/matrix-org/synapse/blob/7dc39858/synapse/logging/context.py#L52) which attempts to fail gracefully when `resource.getrusage(RUSAGE_THREAD)` is unavailable

But that means that when we get [here](https://github.com/matrix-org/synapse/blob/7dc39858/synapse/logging/context.py#L361), `usage_start` is always None, resulting in the warning spam.

**Comment by jryans** [comment]
I assume you want to keep the warnings for platforms that _do_ have support...?

**Comment by jryans** [comment]
IRL answer is yes.

**Comment by jryans** [comment]
I'll give a try given I can clearly reproduce quite easily... 😅

**Comment by richvdh** [comment]
fixed by #6059
